pkg/lib/web/csp.go

```diff
@@ -122,13 +122,25 @@ type dynamicCSPContextKeyType struct{}
 
 var dynamicCSPContextKey = dynamicCSPContextKeyType{}
 
-func WithCSPNonce(ctx context.Context, nonce string) context.Context {
-	return context.WithValue(ctx, dynamicCSPContextKey, nonce)
+type CSPNonceContextValue struct {
+	Nonce string
+}
+
+func WithCSPNonce(ctx context.Context, ctxValue *CSPNonceContextValue) context.Context {
```

Comment:
func WithCSPNonce(ctx context.Context, nonce string) context.Context

The caller need not to know we are using holder pattern.